routes/runtime/github.rb

```diff
@@ -39,8 +39,7 @@ class Clover
       #   for a restore key, the action returns the most recently created cache.
       entry ||= dataset
         .grep(:key, keys.map { |key| "#{DB.dataset.escape_like(key)}%" })
-        .reverse(:created_at)
-        .first
+        .min_by { |e| keys.find_index { |k| e.key.start_with?(k) } + (scopes.index(e.scope) * keys.size) }
```

Comment:
I'm not sure how large the dataset is, and `min_by` would need to iterate over the entire set.  I'll merge this, but I'll try to get this back to the order+first approach.